pyiceberg/io/pyarrow.py

```diff
@@ -2214,13 +2199,16 @@ def _dataframe_to_data_files(
         property_name=TableProperties.WRITE_TARGET_FILE_SIZE_BYTES,
         default=TableProperties.WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT,
     )
+    name_mapping = table_metadata.schema().name_mapping
```

Comment:
Shall we use table's `schema.name-mapping.default` if set and fallback to schema's name-mapping if not?

Reply:
Good question - I'm not sure actually. 

When we are writing a dataframe into an Iceberg table, I think we are making the assumption that its names match the current names of the Iceberg table, so I think using the `table_metadata.schema().name_mapping` is appropriate in supporting that.

`table_metadata.name_mapping` is used to map the field names of written data files hat do not have field IDs to the Iceberg SChema, so I don't think we need to extend our write APIs to also support mapping other names in the name mapping. Since this will be a new feature, I'm of the opinion that we should leave it out until we can think of valid use cases for that from our user base.

I'm curious to hear what others' thoughts are, and whether anyone has a workflow in mind that would benefit from this change!

Reply:
Sounds great! I initially raised this because we’re assigning field IDs for the input dataframe, which aligns with the general purpose of name mapping - to provide fallback IDs. On second thought, schema.name-mapping.default is more for the read side, so using it here may silently introduce unwanted side effects during write. I agree, let’s hold off on this for a while and wait for more discussions.

Reply:
Sounds great 👍  thank you for the review!